subprojects/execution/src/main/java/org/gradle/internal/execution/history/changes/DefaultExecutionStateChangeDetector.java

```diff
@@ -61,10 +62,11 @@ public ExecutionStateChanges detectChanges(AfterPreviousExecutionState lastExecu
             thisExecution.getInputFileProperties(),
             "Input file",
             executable);
-        InputFileChanges directInputFileChanges = new DefaultInputFileChanges(
+        DefaultInputFileChanges allInputFileChanges = new DefaultInputFileChanges(
```

Comment:
This smells weird: we create an object of type `DefaultInputFileChanges`, but we never use it as `InputFileChanges`. Instead, it has two methods that return (potentially new) instances of `DefaultInputFileChanges`. Maybe we should inline what happens in here?

Reply:
Good point! I moved both methods to the change detector.